pkg/controller/sidecarset/sidecarset_utils.go

```diff
@@ -101,3 +112,182 @@ func inconsistentStatus(sidecarSet *appsv1alpha1.SidecarSet, status *appsv1alpha
 		status.UpdatedPods != sidecarSet.Status.UpdatedPods ||
 		status.ReadyPods != sidecarSet.Status.ReadyPods
 }
+
+// add this cache to avoid be influenced by informer cache latency when controller try to count maxUnavailable
+type updatedPodCache struct {
```

Comment:
please consider make updatePodCache an interface and provide a default implementation

Reply:
it's not a good idea to make updatePodCache into a interface at the first place. because nobody knows what will be added into the implemetation.
it will not be refactored into a interface until:
* methods is stable
* other packages need this cache
otherwise I don't think it's necessary to do this change.